src/httpsv_generate.c

```diff
@@ -704,6 +704,22 @@ void HTTPSV_GenerateDemoListing(cluster_t *cluster, oproxy_t *dest)
 	HTTPSV_SendHTMLFooter(cluster, dest);
 }
 
+void HTTPSV_GenerateDemoFilenamesTxt(cluster_t *cluster, oproxy_t *dest)
+{
+	dest->_bufferautoadjustmaxsize_ = 1024 * 1024; // NOTE: this allow 1MB buffer...
+	char row_buf[1024];
+	int i;	
+
+	HTTPSV_SendHTTPHeader(cluster, dest, "200", "text/plain", true);
+
+	Cluster_BuildAvailableDemoList(cluster);
+	for (i = 0; i < cluster->availdemoscount; i++)
```

Comment:
The same check before the pointer dereference should happen here as well.
So we should check if `cluster` is not NULL before the for loop